### ISSUE TITLE
Maven war:war goal fails, e.g., in `mvn package`, if Java/Jakarta EE version of "None" is selected.

### ISSUE THREAD

**Issue by scottkurz** [OpenLiberty/start.openliberty.io#122]
**Describe the bug**  
The war:war goal fails, e.g., in `mvn package`, if a Java/Jakarta EE version of "None" is selected.

**Steps to Reproduce**  
1. Generate Maven starter app with  Java/Jakarta EE version of "None" selected (and some MicroProfile version selection).
2. mvn package

**Expected behavior**  

> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-war-plugin:3.3.2:war (default-war) on project app-name: Error assembling WAR: webxml attribute is required (or pre-existing WEB-INF/web.xml if executing in update mode) -> [Help 1]

**Diagnostic information:**

**Additional context**  

This is clearly explained by the doc: 
https://maven.apache.org/plugins/maven-war-plugin/war-mojo.html#failOnMissingWebXml

It seems like we should explicitly generate the starter with maven-war-plugin config:
`<failOnMissingWebXml>false<failOnMissingWebXml>`

We could either:
* do this just in the case where no Jakarta dep is selected (and therefore no servlet API) OR 
* we could maybe just do this for all cases if that were easier to maintain?

To summarize, I'm saying in the no servlet case at least the war plugin config should look more like:

>                 <plugin>
>                     <groupId>org.apache.maven.plugins</groupId>
>                     <artifactId>maven-war-plugin</artifactId>
>                     <version>3.3.2</version>
>                     <configuration>
>                         <failOnMissingWebXml>false</failOnMissingWebXml>
>                     </configuration>
>                 </plugin>

**Comment by Emily-Jiang** [comment]
I think adding that cofiguration does not hurt.
